lab_3_generate_by_ngrams/main.py

```diff
@@ -159,6 +256,19 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> Optional
 
         In case of corrupt input arguments, None is returned
         """
+        if not isinstance(decoded_corpus, tuple) or len(decoded_corpus) == 0:
+            return None
+
+        text_string = ''
+        decoded_corpus = list(decoded_corpus)
+        decoded_corpus[-1] = '.'
```

Comment:
what if the last element contained something important... like a letter maybe...